src/pybind/mgr/cephadm/serve.py

```diff
@@ -1271,25 +1272,23 @@ async def _create_daemon(self,
                     daemon_spec.name(),
                     ['_orch', 'deploy'],
                     [],
-                    stdin=json.dumps({
-                        "fsid": self.mgr._cluster_fsid,
-                        "name": daemon_spec.name(),
-                        "image": image,
-                        "deploy_arguments": daemon_spec.extra_args,
-                        "params": daemon_params,
-                        "meta": {
-                            'service_name': daemon_spec.service_name,
-                            'ports': daemon_spec.ports,
-                            'ip': daemon_spec.ip,
-                            'deployed_by': self.mgr.get_active_mgr_digests(),
-                            'rank': daemon_spec.rank,
-                            'rank_generation': daemon_spec.rank_generation,
-                            # TODO: Are these two extra_* below redundant?
-                            'extra_container_args': extra_container_args,
-                            'extra_entrypoint_args': extra_entrypoint_args
-                        },
-                        "config_blobs": daemon_spec.final_config,
-                    }),
+                    stdin=exchange.Deploy(
+                        fsid=self.mgr._cluster_fsid,
+                        name=daemon_spec.name(),
+                        image=image,
+                        params=daemon_params,
```

Comment:
so we're fully dropping `deploy_arguments` by the time we get to using the exchange `Deploy` class?

Reply:
Yes. For now that's the case.